video/out/meson.build

```diff
@@ -1,65 +1,52 @@
-wl_protocol_dir = wayland['deps'][2].get_variable(pkgconfig: 'pkgdatadir', internal: 'pkgdatadir')
-protocols = [[wl_protocol_dir, 'stable/presentation-time/presentation-time.xml'],
-             [wl_protocol_dir, 'stable/viewporter/viewporter.xml'],
-             [wl_protocol_dir, 'stable/xdg-shell/xdg-shell.xml'],
-             [wl_protocol_dir, 'unstable/idle-inhibit/idle-inhibit-unstable-v1.xml'],
-             [wl_protocol_dir, 'unstable/text-input/text-input-unstable-v3.xml'],
-             [wl_protocol_dir, 'unstable/linux-dmabuf/linux-dmabuf-unstable-v1.xml'],
-             [wl_protocol_dir, 'unstable/xdg-decoration/xdg-decoration-unstable-v1.xml'],
-             [wl_protocol_dir, 'staging/content-type/content-type-v1.xml'],
-             [wl_protocol_dir, 'staging/fractional-scale/fractional-scale-v1.xml'],
-             [wl_protocol_dir, 'staging/single-pixel-buffer/single-pixel-buffer-v1.xml'],
-             [wl_protocol_dir, 'staging/xdg-activation/xdg-activation-v1.xml']]
-wl_protocols_source = []
-wl_protocols_headers = []
+protocols = [wl_mod.find_protocol('presentation-time'),
+             wl_mod.find_protocol('viewporter'),
+             wl_mod.find_protocol('xdg-shell'),
+             wl_mod.find_protocol('idle-inhibit', state: 'unstable', version: 1),
+             wl_mod.find_protocol('text-input', state: 'unstable', version: 3),
+             wl_mod.find_protocol('linux-dmabuf', state: 'unstable', version: 1),
+             wl_mod.find_protocol('xdg-decoration', state: 'unstable', version: 1),
+             wl_mod.find_protocol('content-type', state: 'staging', version: 1),
+             wl_mod.find_protocol('fractional-scale', state: 'staging', version: 1),
+             wl_mod.find_protocol('single-pixel-buffer', state: 'staging', version: 1),
+             wl_mod.find_protocol('xdg-activation', state: 'staging', version: 1)
+]
 
 foreach v: ['1.32', '1.38', '1.39', '1.41', '1.44']
```

Comment:
Do we still need manual version checks? Maybe we should have named HAVE_ flags per each. For example `HAVE_XDG_ACTIVATION` and so on.

Reply:
Not possible actually, meson hard errors out if the protocol isn't found instead of returning a found() boolean...

I think it's still useful if in the future we need to vendor xml files ourselves, but I'll drop the commit if you think it's not useful

Reply:
It's fine to use it, but it doesn't simplify this much. Maybe we should ask upstream to make it better and return found() and add optional parameter to find_protocol